### PR TITLE
support Ledger Nano S/SP/X devices (blind signing)

### PR DESCRIPTION
**Pull request by yogh333**
support Ledger Nano S/SP/X devices (blind signing)

### CODE REVIEW

**Comment by janek26** [comment]
Bildschirmfoto 2022-06-01 um 11 21 58

Hey, I don't think this has the chance to work this way, as you can see in the error. The background process has no access to the window object and usb methods.
It would need to be it's own content script.

**Review by janek26 — CHANGES_REQUESTED** [review]
most likely this won't get merged until we support multiple account types

**Comment by yogh333** [comment]
Regarding the error it is due to the fact that the first time you add a device you have to launch the extension in a chrome tab (see Expand in settings menu). But you say that there is another solution ?

**Comment by delaaxe** [comment]
given the sensitive nature of a wallet's code base, I'd rather not add a dependency coming from someone's personal github account

**Comment by janek26** [comment]
@yogh333 maybe you can get that under the ledger org?

**Comment by yogh333** [comment]
@janek26 I am working for Ledger with my personal github account. I assume that there are several PRs coming from other github personal accounts so what is the concern here ? Code is open source, can be reviewed, commented, improved, modified etc...

**Comment by yogh333** [comment]
@janek26 [Regarding this issue](https://github.com/argentlabs/argent-x/pull/694#issuecomment-1143352608), did you try by first trying to open the extension in a tab before trying to connect to a Ledger device ?

File: packages/extension/package.json [hunk 1]
```diff
@@ -50,10 +50,12 @@
   "dependencies": {
     "@extend-chrome/messages": "^1.2.2",
     "@google/model-viewer": "^1.11.1",
+    "@ledgerhq/hw-transport-webusb": "^6.27.1",
     "@mui/icons-material": "^5.3.1",
     "@mui/material": "^5.1.0",
     "@mui/styled-engine-sc": "^5.1.0",
     "@tippyjs/react": "^4.2.6",
+    "@yogh/starknetjs-signer-ledger": "^1.1.1",
```
[the remark this answers is not in the document]
Comment (yogh333) [reply packages/extension/package.json:58]: Ok this is an implementation of Starknet JS Signer interface for Ledger device. You mean that you would be more confient if this package come from [LedgerHQ](https://github.com/LedgerHQ) ?

[the remark this answers is not in the document]
Comment (delaaxe) [reply packages/extension/package.json:58]: Yes exactly

[the remark this answers is not in the document]
Comment (yogh333) [reply packages/extension/package.json]: Done ;)

**Comment by yogh333** [comment]
Hello, 
What is the status for the merge ? If there is anything blocking, please tell me so I can help...

**Pull request closed without merging**